Src/AutoNSubstitute/AutoNSubstitute.csproj

```diff
@@ -64,6 +64,7 @@
   <ItemGroup>
     <Compile Include="AutoConfiguredNSubstituteCustomization.cs" />
     <Compile Include="AutoNSubstituteCustomization.cs" />
+    <Compile Include="AutoSubstituteCustomization.cs" />

```

Comment:
> What if the new logic is implemented in the existing `AutoNSubstituteCustomization` and `NSubstituteBuilder` classes instead of defining new classes with similar names?

That might work. What's most important is that we don't introduce breaking changes, or paint ourselves into a corner that we can't get out of.

When it comes to `AutoNSubstituteCustomization`, would it change the overall behaviour, or would it just add a new feature?

Usually, the easiest way to answer that question is to try it and see if anything breaks. If the tests are still green, it's a strong indication that it isn't a breaking change :smile: 

(There may be a few tests that break because they are too tightly coupled to a particular implementation detail, and it *may* be okay to modify such tests, but it's always something we'd need to carefully review. The point with this side remark is that if you try to make the change, and there's one or two failing tests, it may still be OK.)

What if we include the code from  `AutoSubstituteCustomization` into `AutoNSubstituteCustomization` and no tests fail? Could we think of situations where someone else would want that new feature to go away? I can't think of any such situations, but wanted to raise the question for others to ponder...

My gut feeling about this is that it'd probably be fine to merge `AutoSubstituteCustomization` into `AutoNSubstituteCustomization`.

> Do we want all new classes to include `N` in their name?

I think that if a class explicitly uses the NSubstitute API, it should be named accordingly (with the `N`); otherwise, not.